sdks/python/apache_beam/runners/common.py

```diff
@@ -503,6 +534,182 @@ def invoke_process(self,
         windowed_value, self.process_method(windowed_value.value))
 
 
+class _ThreadsafeWatermarkEstimator(object):
```

Comment:
Just a thought: putting all of these in `common.py` requires them to be compiled, and compilation is already a bottleneck (e.g. in testing sometimes we compile it on every worker). Perhaps we could move these to another helper file?

Reply:
Thanks for pointing out that. I'm curious how `common.py` is compiled and how other modules get involved. If moving these into another helper file meanwhile importing the helper file doesn't increase compiling time, then that's the way to go.

Reply:
It's compiled due to https://github.com/apache/beam/blob/release-2.19.0/sdks/python/setup.py#L263 on install.